internal/manifest/v1alpha1/spec_resolver.go

```diff
@@ -0,0 +1,316 @@
+package v1alpha1
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"reflect"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/kyma-project/module-manager/api/v1alpha1"
+	"github.com/kyma-project/module-manager/internal"
+	declarative "github.com/kyma-project/module-manager/pkg/declarative/v2"
+	"github.com/kyma-project/module-manager/pkg/types"
+	"helm.sh/helm/v3/pkg/cli"
+	"helm.sh/helm/v3/pkg/downloader"
+	"helm.sh/helm/v3/pkg/getter"
+	"helm.sh/helm/v3/pkg/repo"
+	"helm.sh/helm/v3/pkg/strvals"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var ErrNoAuthSecretFound = errors.New("no auth secret found")
+
+type ManifestSpecResolver struct {
+	KCP client.Client
+
+	*types.Codec
+	Insecure bool
+
+	ChartCache   string
+	cachedCharts map[string]string
+}
+
+func NewManifestSpecResolver(codec *types.Codec, insecure bool) *ManifestSpecResolver {
+	return &ManifestSpecResolver{
+		Codec:        codec,
+		Insecure:     insecure,
+		ChartCache:   os.TempDir(),
+		cachedCharts: make(map[string]string),
+	}
+}
+
+func (m *ManifestSpecResolver) Spec(ctx context.Context, obj declarative.Object) (*declarative.Spec, error) {
+	manifest, ok := obj.(*v1alpha1.Manifest)
+	if !ok {
+		return nil, fmt.Errorf(
+			"spec resolver can only resolve v1alpha1 Manifests, but was given %s", reflect.TypeOf(obj),
+		)
+	}
+
+	if len(manifest.Spec.Installs) != 1 {
+		return nil, fmt.Errorf("%v installs found in manifest, cannot install", len(manifest.Spec.Installs))
+	}
+
+	install := manifest.Spec.Installs[0]
+
+	specType, err := types.GetSpecType(install.Source.Raw)
+	if err != nil {
+		return nil, err
+	}
+
+	keyChain, err := m.lookupKeyChain(ctx, manifest.Spec.Config, obj.GetNamespace())
+	if err != nil {
+		return nil, err
+	}
+
+	chartInfo, err := m.getChartInfoForInstall(ctx, install, specType, keyChain)
+	if err != nil {
+		return nil, err
+	}
+
+	var mode declarative.RenderMode
+	switch specType {
+	case types.HelmChartType:
+		mode = declarative.RenderModeHelm
+	case types.OciRefType:
+		mode = declarative.RenderModeHelm
+	case types.KustomizeType:
+		mode = declarative.RenderModeKustomize
+	case types.NilRefType:
+		return nil, fmt.Errorf("could not determine render mode for %s", client.ObjectKeyFromObject(manifest))
+	}
+
+	values, err := m.getValuesFromConfig(ctx, manifest.Spec.Config, install.Name, keyChain)
```

Comment:
declarative.Spec.Values can be updated to only handle values.yaml overrides. I know it's `any`, but maybe we can make formatting more specific by using `map[string]any` for example

Reply:
I dont really understand what you mean here, could you elaborate what your goal is?

Reply:
It's related to the [comment](https://github.com/kyma-project/module-manager/pull/203#discussion_r1065524555) above. I meant the config values are of type `any` and a suggestion would be to make it more strict and adherent to only overrides and remove config altogether.

Reply:
the reason why I did this is because kustomize uses a different type and raw does not use them at all. what would we make them more strict on, just map? if so, what would be the benefit?